### PR TITLE
MID:: Add MIDTrending task

### PR DESCRIPTION
**Pull request by ValerieRamillien**

### CODE REVIEW

File: Modules/MUON/MID/src/MIDTrending.cxx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  auto& qcdb = services.get<repository::DatabaseInterface>();
+  trendValues(t, qcdb);
+  generatePlots(qcdb);
+}
+
+void MIDTrending::finalize(Trigger, framework::ServiceRegistryRef services)
+{
+
+  auto& qcdb = services.get<repository::DatabaseInterface>();
+  auto mo = std::make_shared<core::MonitorObject>(mTrend.get(), getName(), "o2::quality_control_modules::trd::MIDTrending", mConfig.detectorName);
```
Comment (knopers8) [inline Modules/MUON/MID/src/MIDTrending.cxx:76]:
```suggestion
  auto mo = std::make_shared<core::MonitorObject>(mTrend.get(), getName(), "o2::quality_control_modules::mid::MIDTrending", mConfig.detectorName);
```
Typo?

File: Modules/MUON/MID/src/MIDTrending.cxx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      auto mo = qcdb.retrieveMO(dataSource.path, dataSource.name, t.timestamp);
+      if (mo == nullptr) {
+        ILOG(Warning, Devel) << "Could not retrieve MO '" << dataSource.name << "' from QCDB, skipping this data source" << ENDM;
+        continue;
+      }
+
+      if (!count) {
+        std::map<std::string, std::string> entryMetadata = mo->getMetadataMap();  // full list of metadata as a map
+        mMetaData.runNumber = std::stoi(entryMetadata[metadata_keys::runNumber]); // get and set run number
+        ntreeentries = (Int_t)mTrend->GetEntries() + 1;
```
Comment (knopers8) [inline Modules/MUON/MID/src/MIDTrending.cxx:101]: Out of curiosity, why do you keep track the number of entries in the tree and store it there, if TTree has `GetEntries`?

**Review by knopers8 — CHANGES_REQUESTED** [review]
It looks correct in general, I only have some quick suggestions.

File: Modules/MUON/MID/src/MIDTrending.cxx (hunk 1 above)
Comment (ValerieRamillien) [reply Modules/MUON/MID/src/MIDTrending.cxx:76]: Yes used TRDTrending as a template for MID and forgot this modification

File: Modules/MUON/MID/src/MIDTrending.cxx (hunk 2 above)
Comment (ValerieRamillien) [reply Modules/MUON/MID/src/MIDTrending.cxx:101]: I did not change this part from the template.

File: Modules/MUON/MID/include/MID/MIDTrending.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  } mMetaData;
+
+  void trendValues(const Trigger& t, repository::DatabaseInterface& qcdb);
+  void generatePlots(repository::DatabaseInterface& qcdb);
+
+  TrendingTaskConfigMID mConfig;
+  Int_t ntreeentries = 0;
+  UInt_t mTime;
+  std::vector<std::string> runlist;
```
[the remark this answers is not in the document]
Comment (ValerieRamillien) [reply Modules/MUON/MID/include/MID/MIDTrending.h:59]: yes I will change. Thanks for you comments

**Review by knopers8 — APPROVED** [review]

**Pull request merged**